index.js

```diff
@@ -1,41 +1,113 @@
 var create = require('webgl-context')
-var getPixels = require('canvas-pixels').get3d
-var triangle = require('a-big-triangle')
 var xtend = require('xtend')
 var assign = require('xtend/mutable')
+var glExt = require('webglew')
+var Framebuffer = require('gl-fbo')
+var Shader = require('gl-shader')
 
-module.exports = function(opt) {
-    opt = xtend({ 
-        width: 1, 
+
+module.exports = function (shader, opt) {
+    if (!opt) {
+        //just options
+        if (typeof shader === 'object' && !shader.fragShader) {
+            opt = shader
+        }
+        //just a shader
+        else {
+            opt = {
+                shader: shader
+            }
+        }
+    }
+    else {
+        opt.shader = shader
+    }
+
+    opt = xtend({
+        width: 1,
         height: 1,
-        preserveDrawingBuffer: true 
+        preserveDrawingBuffer: true,
+        float: true
     }, opt)
-    
-    var gl = opt.gl || create(opt)
+
+    var gl = opt.shader && opt.shader.gl || opt.gl || create(opt)
     if (!opt.shader)
         throw new Error('no shader supplied to gl-shader-output')
-    
+
+    //set gl context dims
+    gl.canvas.width = opt.width
+    gl.canvas.height = opt.height
+
     var shader = typeof opt.shader === 'function'
             ? opt.shader(gl)
             : opt.shader
 
+    //create gl-shader, if only fragment shader source
+    if (typeof shader === 'string') {
+        shader = Shader(gl, '\
+            attribute vec2 position;\
+            void main() {\
+              gl_Position = vec4(position, 1.0, 1.0);\
+            }\
+        ' , shader)
+    }
+
+    //micro optimizations
+    gl.disable(gl.DEPTH_TEST)
+    gl.disable(gl.BLEND)
+    gl.disable(gl.CULL_FACE)
+    gl.disable(gl.DITHER)
+    gl.disable(gl.POLYGON_OFFSET_FILL)
+    gl.disable(gl.SAMPLE_COVERAGE)
+    gl.disable(gl.SCISSOR_TEST)
+    gl.disable(gl.STENCIL_TEST)
+
+    var buffer = gl.createBuffer()
+    gl.bindBuffer(gl.ARRAY_BUFFER, buffer)
+    gl.bufferData(gl.ARRAY_BUFFER, new Float32Array([-1, -1, -1, 3, 3, -1]), gl.STATIC_DRAW)
+    shader.attributes.position.pointer()
+
+    //try to use floats
+    var float = glExt(gl).OES_texture_float && opt.float
```

Comment:
I think we can avoid some complexity of `webglew` here by just using `gl.getExtension`.